drv/lpc55-spi-server/src/main.rs

```diff
@@ -63,74 +369,339 @@ fn main() -> ! {
+            if tctx.rot_irq && tctx.state == TxState::Finish {
+                let (_, _, _, txempty, _, _, _, _, _) = spi.stat();
+
+                if txempty {
+                    ringbuf_entry!(Trace::Line);
+                    // Note: SP is going to be edge-triggered, so this signal
+                    // does not need to be asserted for the entire transaction.
+                    // However, it is nice to see it on the logic analyzer for
+                    // the whole frame.
+                    gpio.set_val(rot_irq, Value::One).unwrap_lite();
+                    ringbuf_entry!(Trace::DeassertRotIrq);
+                    tctx.rot_irq = false;
+                    tctx.state = TxState::Idle;
+                    ringbuf_entry!(Trace::DeassertRotIrq);
+                    // If we have nothing to send in the next frame, then
+                    // the LPC55 will send a byte of unknown origin.
+                    spi.send_u8(0); // write our no-op byte.
+                    spi.drain_tx(); // Throw away byte just written.
+
+                // spi.disable_tx();   // Ignore Tx interrupts.
+                // spi.txerr_clear();
+                } else {
+                    // As long as we are asserting ROT_IRQ, we will be trying
+                    // to keep up with the SP clocking out the Tx FIFO or
+                    // reacting to txempty.
+                    // TODO: We could relax when all our data is in FIFOWR
+                    // and take an interrupt on txempty.
                 }
+            }
 
-                again = true;
+            // This is a slave SPI interface.
+            // The master can always issue a command. It is not obligated
+            // to read our response though one will always be available.
+            // The master might restart or othewise become out of sync at any
+            // time. It may want to start a new session arbitrarily.
+            //
+            // Rx is always enabled at the start of CSn asserted (sot).
+            // Rx will always be read to keep FIFO management simple.
+            // If the first byte indicates a valid protocol, it will be
+            // processed. Protocol errors are reported if they occur.
+            // A 0x00 byte at start will ignore bytes until next start.
+            // A non-zero, non-supported protocol will have an error response.
+            //
+            // If we received the ssd condition, then the read FIFO may still have
+            // several bytes left. Get them now because it is then time to
+            // process the received message.
+            loop {
+                if !spi.has_byte() {
+                    break;
+                }
+                let (b, csn, sot) = spi.read_u8_csn_sot(); // always read
+                ringbuf_entry!(Trace::Rx(rctx.state, b, rctx.count, csn, sot));
+                rctx.rx_byte(b, sot, &mut tctx);
+                if rctx.state == RxState::Header
```

Comment:
Why does this conditional only care if `rctx.state == RxState::Header`? Isn't receiving payload bytes while transmitting _also_ bad?